src/main/java/br/com/ks/viagem/emvoos/controllers/PaisController.java

```diff
@@ -0,0 +1,48 @@
+package br.com.ks.viagem.emvoos.controllers;
+
+import br.com.ks.viagem.emvoos.DTOs.PaisDTO;
+import br.com.ks.viagem.emvoos.services.PaisService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.MediaType;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+import javax.validation.Valid;
+import java.util.List;
+
+import static org.springframework.http.HttpStatus.CREATED;
+
+@RestController
+@RequestMapping("api/v1/pais")
+public class PaisController {
+
+    private PaisService paisService;
+
+
+    @Autowired
+    public PaisController(PaisService paisService) {
+        this.paisService = paisService;
+    }
+
+    @ResponseStatus(CREATED)
+    @PostMapping(value = "/salvar",
+            produces = MediaType.APPLICATION_JSON_VALUE,
+            consumes = MediaType.APPLICATION_JSON_VALUE)
+    public PaisDTO salvarPais(@Valid @RequestBody PaisDTO paisDTO) {
+        return paisService.salvar(paisDTO);
```

Comment:
*Separamos as bordas externas do sistema do seu núcleo. Não ligamos parâmetros de requisição externa com objetos de domínio diretamente.*

Aqui considero que você mandou bem, já que seria bem tranquilo você receber um país direto como argumento aí na hora de salvar. Por outro lado, o nome PaisDTO não denota explicitamente que esse DTO, neste momento é usado para criar um novo país. Inclusive, por conta disso, fica tentador utilizar ele para fazer a listagem também.

*A complicação do nosso é código é proporcional a complicação da nossa feature. Quanto mais simples, melhor.*
Aqui, na minha opinião, você poderia ter utilizado o seu repositório direto. A ideia de ter um Service, segundo o DDD, é você colocar um código que seria complicado de deixar na classe mais rica de seu domínio... O service aqui aumenta o número de indireção do código e me faz ter que navegar por mais classes para entender o fluxo de uma funcionalidade simples. Eu tiraria. 

*Usamos tudo que conhecemos que está pronto. Só fazemos código do zero se for estritamente necessário.*
Quando você configura o starter da aplicação web ele já traz de brinde o jackson como serializador default para parâmetros marcados como RequestBody. Acho que nem precisa colocar o produces e consumes nesse caso. Mas também não é nada demais.

Reply:
Top, sem comentários